src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java

```diff
@@ -0,0 +1,264 @@
+        if (Jenkins.getInstance() != null) {
+            RSAPublicKey publicKey = InstanceIdentity.get().getPublic();
+            msg.setHeader("X-Instance-Identity", Base64.encode(publicKey.getEncoded()));
+        }
+    }
+
+    public static void setInReplyTo(MimeMessage msg, String inReplyTo) throws MessagingException {
+        if (msg == null || inReplyTo == null) {
+            return;
+        }
+        msg.setHeader("In-Reply-To", inReplyTo);
+        msg.setHeader("References", inReplyTo);
+    }
+
+    public static interface AddressFilter {
+        Set<InternetAddress> apply(Set<InternetAddress> recipients);
+    }
+
+    private void addSubject(MimeMessage msg) throws MessagingException {
+        if (subject != null) {
+            msg.setSubject(subject);
+        }
+    }
+
+    private void addBody(MimeMessage msg) throws MessagingException {
+        if (body != null) {
+            Multipart multipart = new MimeMultipart();
+            BodyPart bodyPart = new MimeBodyPart();
+
+            bodyPart.setContent(body, mimeType);
+            multipart.addBodyPart(bodyPart);
+            msg.setContent(multipart);
+        }
+    }
+
+    private void addRecipients(MimeMessage msg) throws UnsupportedEncodingException, MessagingException {
+        addRecipients(msg, to, Message.RecipientType.TO);
+        addRecipients(msg, cc, Message.RecipientType.CC);
+        addRecipients(msg, bcc, Message.RecipientType.BCC);
+    }
+    private void addRecipients(MimeMessage msg, Set<InternetAddress> recipientList, Message.RecipientType recipientType) throws UnsupportedEncodingException, MessagingException {
+        if (recipientList.isEmpty()) {
+            return;
+        }
+
+        if (recipientFilter != null) {
+            Set<InternetAddress> filteredList = recipientFilter.apply(recipientList);
+            msg.setRecipients(recipientType, filteredList.toArray(new InternetAddress[filteredList.size()]));
+        } else {
+            msg.setRecipients(recipientType, recipientList.toArray(new InternetAddress[recipientList.size()]));
+        }
+    }
+
+    private InternetAddress toNormalizedAddress(String address) throws UnsupportedEncodingException {
+        if (address == null) {
+            return null;
+        }
+
+        // if not a valid address (i.e. no '@'), then try adding suffix
+        if (!address.contains("@")) {
+            if (defaultSuffix != null && defaultSuffix.contains("@")) {
+                address += defaultSuffix;
+            } else {
+                return null;
+            }
+        }
+
+        try {
+            return Mailer.StringToAddress(address, charset);
+        } catch (AddressException e) {
+            // report bad address, but try to send to other addresses
+            if (listener != null) {
+                listener.getLogger().println("Unable to send to address: " + address);
+                e.printStackTrace(listener.error(e.getMessage()));
```

Comment:
Can be simplified to:

    e.printStackTrace(listener.error("Unable to send to address: " + address));